File: migration/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{

    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while(change < 0); // prompting user for a positive number
    
    // convert float to int, rounding to the nearest penny
    int cents = round(change * 100);
    // initialize counter to keep track of the number of coins
    int counter = 0;
    while(cents > 0)
    {
        if(cents >= 25)
        {
            while(cents >= 25)
            {
                cents -= 25;
                counter++;
            }
        }
        else if(cents >= 10)
        {
            while(cents >= 10)
            {
                cents -=10;
                counter++;
            }
        }
        else if(cents >= 5)
        {
            while(cents >= 5)
            {
                cents -= 5;
                counter++;
            }
        }
        else if(cents >= 1)
        {
            while(cents >= 1)
            {
                cents -= 1;
                counter++;
            }
        }
    }
    
    printf("%i\n", counter);

}

File: migration/pset6/cash/cash.py
# while True, try/except to emulate do/while
while True:
    try:
        change = float(input("Change owed: "))
    except ValueError:
        continue
    if change >= 0:
        break
    
cents = int(round(change * 100))

counter = 0
while cents > 0:
    # while the number >= 25, decrease by 25 until it is < 25
    # then try the next denomination, and so on
    if cents >= 25:
        while cents >= 25:
            cents -= 25
            counter += 1
    elif cents >= 10:
        while cents >= 10:
            cents -= 10
            counter += 1
    elif cents >= 5:
        while cents >= 5:
            cents -= 5
            counter += 1
    elif cents >= 1:
        while cents >= 1:
            cents -= 1
            counter += 1

print(counter)
    